consensus/safety-rules/src/persistent_safety_storage.rs

```diff
@@ -217,4 +225,48 @@ mod tests {
         assert_eq!(safety_data.last_voted_round, 8);
         assert_eq!(safety_data.preferred_round, 1);
     }
+
+    #[test]
+    fn test_waypoint_counters() {
+        let consensus_private_key = ValidatorSigner::from_int(0).private_key().clone();
+        let storage = Storage::from(InMemoryStorage::new());
+        let mut safety_storage = PersistentSafetyStorage::initialize(
+            storage,
+            Author::random(),
+            consensus_private_key,
+            Ed25519PrivateKey::generate_for_testing(),
+            Waypoint::default(),
+            true,
+        );
+
+        let waypoint = safety_storage.waypoint().unwrap();
+        assert_eq!(
+            counters::get_state(counters::WAYPOINT_VERSION) as u64,
+            waypoint.version()
+        );
```

Comment:
Ah, you want it for a test.

If you look elsewhere in the code, we do features for testing.  Here's an example https://github.com/diem/diem/blob/6c36ab4b32b20069147caf1b671493e76d8097f0/crypto/crypto/src/test_utils.rs#L79 where it will only be built for test builds.

Reply:
Thanks!! This is something new I learnt today!